engine/graphics/image.cpp

```diff
@@ -2,55 +2,62 @@
 
 #include "utils.h"
 
-namespace pong
-{
-
-namespace image
+namespace pong::image
 {
 
 /* Converts an alpha image to RGBA format. */
-std::vector<unsigned char> ConvertAlphaImageToRGBA(const std::vector<unsigned char>& imageData)
+Image ConvertAlphaImageToRGBA(const Image& image)
 {
-    std::vector<unsigned char> convertedImage {};
-    convertedImage.reserve(imageData.size() * 4);
+    ASSERT(image.mComponents == 1);
+
+    Image convertedImage {};
+    convertedImage.mPixels.reserve(image.mPixels.size() * 4);
 
     const unsigned char kFullColor = 255;
-    for (auto& alpha : imageData)
+    for (auto& alpha : image.mPixels)
     {
-        convertedImage.emplace_back(kFullColor);
-        convertedImage.emplace_back(kFullColor);
-        convertedImage.emplace_back(kFullColor);
-        convertedImage.emplace_back(alpha);
+        convertedImage.mPixels.push_back(kFullColor);
+        convertedImage.mPixels.push_back(kFullColor);
+        convertedImage.mPixels.push_back(kFullColor);
+        convertedImage.mPixels.push_back(alpha);
     }
 
+    convertedImage.mWidth = image.mWidth;
+    convertedImage.mHeight = image.mHeight;
+    convertedImage.mComponents = 4;
+
     return convertedImage;
 }
 
 /* Flips an image so top left element becomes bottom left element. */
-std::vector<unsigned char> FlipImageVertically(const std::vector<unsigned char>& imageData, int width, int height, int comp)
+Image FlipImageVertically(const Image& image)
```

Comment:
Much better. you see how this removed complexity and made the code easier to change in the future? now if you support 3d images (for some reason) you don't have to edit the interface. If you change how width is calculated, you can do that internally to the Image class and no one is the wiser